components/withBreadcrumbs.tsx

```diff
@@ -20,11 +21,38 @@ const WithBreadcrumbs: FC = () => {
       return [];
     }
 
-    return getSideNavigation([navigationKey as NavigationKeys])
-      .map(([, item]) => item.items)
-      .flat()
-      .filter(([, item]) => pathname.includes(item.link))
-      .map(([, item]) => ({ label: item.label, href: item.link }));
+    const navigationTree = getSideNavigation([navigationKey as NavigationKeys]);
+
+    const toCamelCase = (str: string) =>
```

Comment:
As mentioned before, please extract this to the `stringUtils` utility 👀 this shouldn't be here. You can also achieve the same with regex. Here's an optimised version that works for any sort of given string:

```ts
const toCamelCase = (str: string) =>
  str
    .replace(/[^a-zA-Z0-9]+(.)/g, (m, chr) => chr.toUpperCase())
    .replace(/^[A-Z]/, chr => chr.toLowerCase());
```